tests/core/test_otel_logger.py

```diff
@@ -320,3 +333,43 @@ def test_timer_start_and_stop_manually_send_true(self, mock_time, name):
         self.meter.get_meter().create_observable_gauge.assert_called_once_with(
             name=full_name(name), callbacks=ANY
         )
+
+    def test_get_name_invalid_cases(self):
```

Comment:
Non-blocking nit-pick:  All of these expected-to-fail edge cases could be one parameterized test which parameterizes the name and the expected error message.